src/pymor/discretizers/builtin/cg.py

```diff
@@ -1108,12 +1127,18 @@ def discretize_stationary_cg(analytical_problem, diameter=None, domain_discretiz
             for i, rd in enumerate(p.robin_data[0].functions):
                 robin_tuple = (rd, p.robin_data[1])
                 Li += [RobinBoundaryOperator(grid, boundary_info, robin_data=robin_tuple, name=f'robin_{i}')]
+                if energy_product:
+                    eLi += [RobinBoundaryOperator(grid, boundary_info, robin_data=robin_tuple)]
             coefficients += list(p.robin_data[0].coefficients)
         else:
             Li += [RobinBoundaryOperator(grid, boundary_info, robin_data=p.robin_data, name=f'robin')]
+            if energy_product:
+                eLi += [RobinBoundaryOperator(grid, boundary_info, robin_data=p.robin_data)]
             coefficients.append(1.)
 
     L = LincombOperator(operators=Li, coefficients=coefficients, name='ellipticOperator')
+    if energy_product:
+        eL = LincombOperator(operators=eLi, coefficients=coefficients, name='ellipticEnergyProduct')
```

Comment:
I think that `coefficients` is wrong for the affinely decomposed case. Probably just `[1.] * len(eLi)`?

Reply:
No, I think in general this should include the evaluation of the coefficients, otherwise I dont see how `op.assemble(mu) == energy_product`. I guess the easiest is to revert to adding each component of the diffusion separately...

Reply:
> No, I think in general this should include the evaluation of the coefficients

This happens when `p.diffusion` or `p.reaction` is evaluated ..

Reply:
You are right, of course.